data_structures/inorder.cpp

```diff
@@ -0,0 +1,181 @@
+ * @author [Chaitanya Giri](https://github.com/Chaitanya134)
+ */
+
+#include <iostream>  /// for IO operations
+#include <cassert>   /// for assert
+#include <queue>     /// for storing node in level order traversal
+#include <sstream>   /// for converting expected output to a string
+
+/**
+ * @namespace
+ * @brief Data structures
+ */
+namespace data_structures {
+
+/**
+ * @class Node
+ * @brief Represents a node in a binary tree.
+ * 
+ * This class stores an integer value and pointers to the left and right child nodes.
+ */
+
+class Node {
+public:
+    int data = 0;    ///< data stored in this node
+    Node *left;      ///< pointer to the left node
+    Node *right;     ///< pointer to the right node
+};
+
+/**
+ * @class BinaryTree
+ * @brief Represents a binary tree.
+ * 
+ * This class provides operations for inserting nodes into the tree and traversing the nodes in
+ * order.
+ */
+class BinaryTree {
+public:
+    /**
+    * @brief insert a node in the tree using level order traversal
+    * @param root root node of the tree
+    * @param val value to be inserted
+    * @returns void
+    */
+    void insertNode(Node* &root, int val) {
+        // create a new node
+        Node *newNode = new Node;
+        newNode->data = val;
+        newNode->left = nullptr;
+        newNode->right = nullptr;
+
+        if (root == nullptr) {
+            // point root to the new node if root is NULL
+            root = newNode;
+            return;
+        }
+
+        // queue for level order traversal
+        std::queue<Node*> q;
+        q.push(root);
+
+        while (!q.empty()) {
+            Node *node = q.front();
+            q.pop();
+
+            if (node->left) {
+                // add to left child to queue
+                q.push(node->left);
+            } else {
+                // add new node to the left if no left child is present
+                node->left = newNode;
+                return;
+            }
+
+            if (node->right) {
+                // add right child to queue
+                q.push(node->right);
+            } else {
+                // add new node to the right if no right child is present
+                node->right = newNode;
+                return;
+            }
+        }
+    }
+
+    /**
+    * @brief [Algorithm implementation for inorder tree traversal]
+    * @param node node from which inorder traversal will start
+    * @returns void
+    */
+    void inorderTraversal(Node *node) {
+        if (node == nullptr) return;       // stop traversal if node is NULL
+
+        inorderTraversal(node->left);   // recursively traverse current node's left subtree
+        std::cout << node->data << " ";      // print the current node's data
+        inorderTraversal(node->right);  // recursively traverse current node's right subtree
+    }
+};
+} // namespace data_structures
+
+/**
+* @brief [Self-tests for inorder tree traversal]
```

Comment:
There's no need for the brackets. That's usually used only for Markdown links.
```suggestion
* @brief Self-tests for inorder tree traversal
```